### PR TITLE
[Tradeshield]: Handle error and improve execution logs

### PR DESCRIPTION
**Pull request by amityadav0**
## Description

- Handle error and improve execution logs

### What has Changed?

What specific problem were you aiming to address, and how did you successfully resolve it? If tests were not uploaded for this pull request or if coverage decreased, please provide an explanation for the change.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/elys-network/elys/blob/main/docs/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#testing)
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### Deployment Notes

Are there any specific considerations to take into account when deploying these changes? This may include new dependencies, scripts that need to be executed, or any aspects that can only be evaluated in a deployed environment.

### Screenshots and Videos

Please provide any relevant before and after screenshots by uploading them here. Additionally, demo videos can be highly beneficial in demonstrating the process.

### CODE REVIEW

File: x/tradeshield/keeper/msg_server_execute_orders.go [hunk 1]
```diff
@@ -67,10 +72,17 @@ func (k msgServer) ExecuteOrders(goCtx context.Context, msg *types.MsgExecuteOrd
...
-			return nil, err
+			// Add log about error or not executed
+			perpLog = append(perpLog, fmt.Sprintf("Perpetual order Id:%d cannot be executed due to err: %s", perpetualOrderId, err.Error()))
 		}
 	}
 
+	ctx.EventManager().EmitEvent(sdk.NewEvent(types.TypeEvtExecuteOrders,
+		sdk.NewAttribute("spot_orders", strings.Join(spotLog, "\n")),
+		sdk.NewAttribute("perpetual_orders", strings.Join(perpLog, "\n")),
+	))
```
Comment (cosmic-vagabond) [inline x/tradeshield/keeper/msg_server_execute_orders.go:85]: @amityadav0 can you add unit tests that confirm that given the edge case scenarios we are getting those logs as event attributes fired?

**Review by cosmic-vagabond — CHANGES_REQUESTED** [review]

**Comment by amityadav0** [comment]
Testing results:
```
ay@Amits-MBP ~ % elysd tx tradeshield execute-orders 1 1 --from=elys1dd34v384hdfqgajkg0jzp0y5k6qlvhltt76qd5 --gas=400000

- attributes:
  - index: true
    key: spot_orders
    value: 'Spot order Id:1 cannot be executed due to err: price not found'
  - index: true
    key: perpetual_orders
    value: ""
  - index: true
    key: msg_index
    value: "0"
  type: execute_orders
```
Tx was successful and error logs were added

**Review by cosmic-vagabond — APPROVED** [review]
